src/base/static/components/organisms/main-map.js

```diff
@@ -0,0 +1,377 @@
+import { Component } from "react";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+
+import MapboxGLProvider from "../../libs/maps/mapboxgl-provider";
+// Import other providers here as they become available
+
+import emitter from "../../utils/emitter";
+import { createGeoJSONFromCollection } from "../../utils/collection-utils";
+
+import { mapConfigSelector } from "../../state/ducks/map-config";
+import {
+  mapLayersBasemapSelector,
+  mapSizeValiditySelector,
+  setMapSizeValidity,
+  setMapPosition,
+} from "../../state/ducks/map";
+
+import constants from "../../constants";
+
+const Util = require("../../js/utils.js");
+
+class MainMap extends Component {
+  constructor(props) {
+    super(props);
+    this._router = props.router;
+    this._places = props.places;
+    this._layers = props.mapConfig.layers;
+
+    let MapProvider;
+    switch (props.mapConfig.provider) {
+      // Add other provider types here as they become available
+      case "mapboxgl":
+        MapProvider = MapboxGLProvider;
+        break;
+      default:
+        MapProvider = MapboxGLProvider;
+        break;
+    }
+
+    // Bind visiblity event for custom layers
+    $(Shareabouts).on("visibility", async (evt, layer) => {
+      await this.createLayerFromConfig(layer);
+    });
+
+    this._map = MapProvider(props.container, props.mapConfig.options);
+
+    // TEMPORARY: Manually trigger the visibility of layers for testing
+    this._map.on("load", () => {
+      $(Shareabouts).trigger("visibility", [this._layers[4], true, true]);
+    });
+
+    if (props.mapConfig.geolocation_enabled) {
+      // TODO
+    }
+  }
+
+  componentDidUpdate() {
```

Comment:
Here's how we're handling the size invalidation for the time being.

Reply:
Yeah, it's a bit janky for now, but it's not a bad workaround for now.

Just to recap our convo from earlier, I think it should eventually look like this:

```js
componentDidUpdate(prevProps) {
  const shouldTriggerMapRecenter =
    prevProps.isContentPanelOpen !== this.props.isContentPanelOpen ||
    prevProps.isStorySidebarOpen !== this.props.isStorySidebarOpen
  if (shouldTriggerMapRecenter) {
    this._map.invalidateSize()
  }
}
```